docs/Contributing/pull-request.rst

```diff
@@ -0,0 +1,182 @@
+How to Submit a Pull Request
+****************************
+
+A Pull Request (PR), also commonly known as a Merge Request, is a mechanism to "pull" code from one repository
+into another one and merge the change into the source repository.
+
+Before we go much further, let's take a few minutes to describe what happened when you :doc:`dev-env` and to
+define a few terms that will help with futher instructions. Starting with a diagram of the setup of the three
+locations where code resides:
+
+.. image:: ../_static/images/github_repos.png
+    :align: center
+    :height: 300
+    :alt: Diagram of GitHub Repositories
```

Comment:
I think this entire discussion is a bit too abstract. The git docs will give the general (abstract) picture, what we need to add is the expected workflow with Github. Some suggestions:

* Add icons to the diagram (Github logo, personal computer) and add labels `longturn/freeciv21`, `johndoe/freeciv21`, "your computer"
* The typical workflow is to:
    1. Get the latest code from upstream
    2. Work on it
    3. Push to your fork
    4. Make pull request
    
  This could be emphasized by adding numbers to the diagram and a circular arrow from/to the local.
* No need to emphasize bidirectional communication between the fork and the local. All code updates should always come from the central repo anyway (unless you're testing on another machine and such advanced use cases).

Reply:
simplified in next commit